src/main/java/org/opensearch/neuralsearch/processor/combination/ArithmeticMeanScoreCombinationMethod.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import java.util.Objects;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Abstracts combination of scores based on arithmetic mean method
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ArithmeticMeanScoreCombinationMethod implements ScoreCombinationMethod {
+
+    private static ArithmeticMeanScoreCombinationMethod INSTANCE = new ArithmeticMeanScoreCombinationMethod();
```

Comment:
In general this may be true, but in this case the only place we use it is the enum ```ScoreCombinationTechnique``` members, I doubt mock are gonna be used there.

Reply:
Actually, I do not favor having this instance in the enum. We can discuss about it in another thread.